src/main/java/Duke.java

```diff
@@ -1,10 +1,261 @@
+    public static String home = System.getProperty("user.home");
+    //    public static Path filePath = java.nio.file.Paths.get(home, "ip", "task_list.txt");
+    public static String filePath = "./task_list.txt";
+
+    public static void createFile(String filePath) {
+        try {
+            File f = new File(filePath);
+            if (f.createNewFile()) {
+                System.out.println("File created: " + f.getName());
+            } else {
+                System.out.println("File already exists");
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred while creating file.");
+            e.printStackTrace();
+        }
+    }
+
+    public static void readTasksFromFile(File f) throws FileNotFoundException {
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String task = s.nextLine();
+            String[] taskElements = task.trim().split("\\|");
+            String taskType = taskElements[0].trim();
+            boolean taskStatus = taskElements[1].trim().equals("X");
+            String taskDescription = taskElements[2].trim();
+            switch (taskType) {
+            case "T":
+                tasks.add(new ToDo(taskDescription, taskStatus));
+                break;
+            case "D":
+                String deadlineDate = taskElements[3].trim();
+                tasks.add(new Deadline(taskDescription, taskStatus, deadlineDate));
+                break;
+            case "E":
+                String startDate = taskElements[3].trim();
+                String endDate = taskElements[4].trim();
+                tasks.add(new Event(taskDescription, taskStatus, startDate, endDate));
+                break;
+            default:
+            }
+        }
+    }
+
+    /*
+     T | X | description
+     D |   | description | time
+     E |   | description | start time | end time
+     */
+
+    public static void saveTasksToFile(String filePath) throws IOException {
+        FileWriter fw = new FileWriter(filePath, false);
+        for (Task t : tasks) {
+            String completeTaskDescription = t.printTask();
+            String taskType = completeTaskDescription.substring(1, 2);
+            switch (taskType) {
+            case "T":
+                fw.write(taskType + " | " + t.getStatusIcon() + " | " + t.getDescription() +
+                        System.lineSeparator());
+                break;
+            case "D":
+                fw.write(taskType + " | " + t.getStatusIcon() + " | " + t.getDescription() + " | " +
+                        ((Deadline) t).getDeadline() + System.lineSeparator());
+                break;
+            case "E":
+                fw.write(taskType + " | " + t.getStatusIcon() + " | " + t.getDescription() + " | " +
+                        ((Event) t).getStartTime() + " | " + ((Event) t).getEndTime() + System.lineSeparator());
+                break;
+            default:
+            }
```

Comment:
Perhaps you can consider abstracting this block of code to a separate function in order to minimise deeply-nested code and improve code readability.